medical/models/medical_patient.py

```diff
@@ -1,105 +1,114 @@
+            record.age = years_months_days
 
-    @api.one
-    def action_invalidate(self):
-        self.active = False
-        self.partner_id.active = False
+    @api.multi
+    def _compute_is_deceased(self):
+        for record in self:
+            record.is_deceased = bool(record.date_death)
 
-    age = fields.Char(compute='_compute_age')
-    identification_code = fields.Char(
-        string='Internal Identification',
-        help='Patient Identifier provided by the Health Center.'
-        '(different from the Social Security Number)')
-    general_info = fields.Text(string='General Information')
-    dob = fields.Date(string='Date of Birth')
-    dod = fields.Datetime(string='Deceased Date')
-    active = fields.Boolean(default=True)
-    deceased = fields.Boolean()
-    partner_id = fields.Many2one(
-        comodel_name='res.partner', required=True, ondelete='cascade',
-        index=True)
-    gender = fields.Selection(
-        selection=[
-            ('m', 'Male'),
-            ('f', 'Female'),
-        ])
-    medical_center_id = fields.Many2one(
-        comodel_name='res.partner',
-        domain="[('is_institution', '=', True)]",
-        string='Medical Center'
-    )
-    marital_status = fields.Selection(
-        selection=[
-            ('s', 'Single'),
-            ('m', 'Married'),
-            ('w', 'Widowed'),
-            ('d', 'Divorced'),
-            ('x', 'Separated'),
-            ('z', 'law marriage'),
-        ])
+    @api.multi
+    @api.constrains('is_pregnant', 'gender')
+    def _check_is_pregnant(self):
+        for record in self:
+            if record.is_pregnant and record.gender != 'female':
+                raise ValidationError(_(
+                    'Invalid selection - Only a `Female` may be pregnant.',
+                ))
 
     @api.model
-    @api.returns('self', lambda value: value.id)
-    def create(self, vals):
-        vals['is_patient'] = True
+    def _create_vals(self, vals):
+        vals = super(MedicalPatient, self)._create_vals(vals)
         if not vals.get('identification_code'):
-            sequence = self.env['ir.sequence'].get('medical.patient')
-            vals['identification_code'] = sequence
-        return super(MedicalPatient, self).create(vals)
+            Seq = self.env['ir.sequence']
+            vals['identification_code'] = Seq.next_by_code(
+                self._name,
+            )
+        vals.update({
+            'customer': True,
+        })
+        return vals
+
+    @api.model
+    def _get_default_image(self, vals):
+        res = super(MedicalPatient, self)._get_default_image(vals)
+        if res:
+            return res
+        img_path = get_module_resource(
+            'medical', 'static/src/img', 'patient-avatar.png',
+        )
+        with open(img_path, 'r') as image:
+            base64_image = image.read().encode('base64')
+            return tools.image_resize_image_big(base64_image)
```

Comment:
At this step we are done with `image` already. Shouldn't the `return` be outside the `with`?